src/MorseL/HubWebSocketHandler.cs

```diff
@@ -152,27 +157,30 @@ private static bool IsHubMethod(MethodInfo methodInfo)
 
         public override async Task ReceiveAsync(Connection connection, string serializedInvocationDescriptor)
         {
-            var invocationDescriptor = Json.DeserializeInvocationDescriptor(serializedInvocationDescriptor, _methods.Values.Select(d => d.MethodExecutor.MethodInfo).ToArray());
+            InvocationDescriptor invocationDescriptor = null;
+            try
+            {
+                invocationDescriptor = Json.DeserializeInvocationDescriptor(serializedInvocationDescriptor, _methods.Values.Select(d => d.MethodExecutor.MethodInfo).ToArray());
+            }
+            catch (Exception e)
+            {
+                await HandleUnparseableReceivedInvocationDescriptor(connection, serializedInvocationDescriptor, e);
+                return;
+            }
 
             if (invocationDescriptor == null)
             {
-                await connection.Channel.SendMessageAsync(new Message()
-                {
-                    MessageType = MessageType.Text,
-                    Data = $"Cannot find method to match inbound request for {connection.Id}"
-                }).ConfigureAwait(false);
+                // Valid JSON but unparseable into a known, typed invocation descriptor (unknown method name, invalid parameters)
+                await HandleInvalidReceivedInvocationDescriptor(connection, serializedInvocationDescriptor);
                 return;
             }
 
             HubMethodDescriptor descriptor;
             if (!_methods.TryGetValue(invocationDescriptor.MethodName, out descriptor))
             {
-                await connection.Channel.SendMessageAsync(new Message()
-                {
-                    MessageType = MessageType.Text,
-                    Data = $"Cannot find method {invocationDescriptor.MethodName} to match inbound request for {connection.Id}"
-                }).ConfigureAwait(false);
-                return;
+                // Really strange and unlikely, valid JSON and was known but not found here
+                // Likely not possible
+                await HandleMissingReceivedInvocationDescriptor(connection, invocationDescriptor);
```

Comment:
I definitely appreciate the explicit error handling here. Compared to the previous "just send a message" this is a substantial improvement.